content/en/blog/2022/cryptomb-privatekeyprovider/index.md

```diff
@@ -0,0 +1,226 @@
+
+### Istio Mesh wide Configuration
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  namespace: istio-system
+  name: example-istiocontrolplane
+spec:
+  profile: demo
+  components:
+    egressGateways:
+    - name: istio-egressgateway
+      enabled: true
+    ingressGateways:
+    - name: istio-ingressgateway
+      enabled: true
+  meshConfig:
+    defaultConfig:
+      privateKeyProvider:
+        cryptomb:
+          pollDelay: 10ms
+{{< /text >}}
+
+### Istio Gateways Configuration
+
+If a user wants to apply a private key provider configuration for ingress gateway only, follow the below sample configuration.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  namespace: istio-system
+  name: example-istiocontrolplane
+spec:
+  profile: demo
+  components:
+    egressGateways:
+    - name: istio-egressgateway
+      enabled: true
+    ingressGateways:
+    - name: istio-ingressgateway
+      enabled: true
+      k8s:
+        podAnnotations:
+          proxy.istio.io/config: |
+            privateKeyProvider:
+              cryptomb:
+                pollDelay: 10ms
+{{< /text >}}
+
+### Istio Sidecar Configuration using pod annotations
+
+If a user wants to apply private key provider configuration to application specific pods, configure them using pod annotations like the below sample.
+
+{{< text yaml >}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: httpbin
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: httpbin
+  labels:
+    app: httpbin
+    service: httpbin
+spec:
+  ports:
+  - name: http
+    port: 8000
+    targetPort: 80
+  selector:
+    app: httpbin
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: httpbin
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: httpbin
+      version: v1
+  template:
+    metadata:
+      labels:
+        app: httpbin
+        version: v1
+      annotations:
+        proxy.istio.io/config: |
+          privateKeyProvider:
+            cryptomb:
+              pollDelay: 10ms
+    spec:
+      serviceAccountName: httpbin
+      containers:
+      - image: docker.io/kennethreitz/httpbin
+        imagePullPolicy: IfNotPresent
+        name: httpbin
+        ports:
+        - containerPort: 80
+{{< /text >}}
+
+### Performance
+
+The potential performance benefit depends on many factors. For example, the size of the cpuset Envoy is running on, incoming traffic pattern, encryption type (RSA or ECDSA), and key size. Here is a white paper with [performance](https://www.intel.com/content/www/us/en/architecture-and-technology/crypto-acceleration-in-xeon-scalable-processors-wp.html) numbers.
+
+Istio ingress gateway TLS handshake performance comparison on 1.14-dev on May 10th 2022.
+
+{{< image link="./istio-ingress-gateway-tls-handshake-perf-num.png" caption="Istio ingress gateway TLS handshake performance comparison" >}}
```

Comment:
40ms is extremely higher than expected (order of magnitude). Are you sure the test was configured correctly? Do you have more details on the setup?

Is most of the time spent on network latency from k6 -> cluster?

Reply:
yes, this is a total latency including k6- > gateway- >fortio server, so it is higher, We will submit a PR to clearly state this out and our detailed settings.